frontend/src/views/MyDrugsPage/MyDrugsPage.jsx

```diff
@@ -1,21 +1,88 @@
 import './MyDrugsPage.scss';
 import '../../components/Pill/Pill.module.scss';
 
-import { Button, Typography } from '@mui/material';
-import React, { useState } from 'react';
+import { Box, Button, Modal, Typography } from '@mui/material';
+import React, { memo, useState } from 'react';
 
+import DrugModal from '../../components/AddOrUpdateDrug/DrugModal';
 import buttonStyles from '../../components/Button/Button.module.scss';
 import { AID_KIT_IMAGE_PATH } from '../../constants/images';
 import drugs from '../../mock/drugs';
 
 const initialDescription = 'CHOOSE A DRUG TO SHOW THE DESCRIPTION';
 const initialImageSource = AID_KIT_IMAGE_PATH;
 
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: '90vw',
+  maxWidth: '700px',
+  minHeight: '450px',
+  bgcolor: 'background.default',
+  border: '2px solid #000',
+  borderRadius: '20px',
+  boxShadow: 24,
+  p: 4,
+  mt: 3,
+};
+
 const MyDrugsPage = () => {
   const [description, setDescription] = useState(initialDescription);
   const [photoSource, setPhotoSource] = useState(initialImageSource);
+  const [drugSelected, setDrugSelected] = useState(null);
+  const [open, setOpen] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('');
+
+  const handleClose = () => setOpen(false);
+
+  const ShowUpdateButton = memo(({ isDescription }) => {
+    if (isDescription !== initialDescription) {
+      return <DrugModal drugAction={patchMedication} actionName="Update" />;
+    }
+    return <div />;
+  });
+
+  // At this moment works only if one of mock medications has the same id as any in the collection, because there is no functionality for getting all the user's medications.
+  async function patchMedication(drug) {
+    const drugName = drugSelected;
+    var id;
+    var updatedDrug;
+    drugs.forEach((medication) => {
+      if (medication.name.toUpperCase() === drugName) {
+        updatedDrug = medication;
+        id = medication.id;
+      }
+    });
+    drug.id = id;
+    drug.addDate = updatedDrug.addDate;
+    const requestBody = {
+      nameOfMedication: drug.name,
+      quantity: drug.quantity,
+      addDate: drug.addDate,
+      dosage: drug.dosage,
+      expirationDate: drug.expirationDate,
+    };
+    try {
+      const response = await fetch(`http://localhost:8080/api/medications/${id}`, {
```

Comment:
please create a constant with the baseline of the URL (http://localhost:8080/) it will be used in many places but it will change if server is in the different environment